### PR TITLE
Provide support for dns_cache_config in addons

### PR DESCRIPTION
**Pull request by sundarsudandiraraj**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -150,6 +150,12 @@ variable "enable_addon_horizontal_pod_autoscaling" {
   default     = true
 }
 
+variable "enable_addon_dns_cache_config" {
+  description = "Whether to enable NodeLocal DNSCache addon. NodeLocal DNSCache improves DNS lookup latency, makes DNS lookup times more consistent, and reduces the number of DNS queries to kube-dns by running a DNS cache on each node in a cluster. It is disabled by default. See: https://cloud.google.com/kubernetes-engine/docs/how-to/nodelocal-dns-cache"
+  type        = bool
+  default     = false
```
Comment (shardulsrivastava) [inline variables.tf:156]: As a best practice, we should enable this by default.

File: main.tf [hunk 2]
```diff
@@ -126,6 +127,9 @@ resource "google_container_cluster" "k8s_cluster" {
     horizontal_pod_autoscaling {
       disabled = ! var.enable_addon_horizontal_pod_autoscaling
     }
+    dns_cache_config {
```
Comment (shardulsrivastava) [inline main.tf:130]: you can add a link to `NodeLocal DNSCache` here - 
https://cloud.google.com/kubernetes-engine/docs/how-to/nodelocal-dns-cache

**Review by shardulsrivastava — CHANGES_REQUESTED** [review]
Added a few comments

File: main.tf (hunk 2 above)
Comment (sundarsudandiraraj) [reply main.tf:130]: I have added this in the Variable's description. Shouldn't that be enough?

File: variables.tf (hunk 1 above)
Comment (sundarsudandiraraj) [reply variables.tf:156]: yeah, we can. Only disabling this will results in a disruptive operation.

**Review by syedrakib — APPROVED** [review]
LGTM

**Review by shardulsrivastava — APPROVED** [review]
LGTM

**Pull request merged**